tests/common.go

```diff
@@ -1970,11 +2133,171 @@ func updateNamespace(in interface{}, namespaceMapping map[string]string) error {
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*batchv1beta1.CronJob); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*batchv1.Job); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*corev1.LimitRange); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*networkingv1beta1.Ingress); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*monitoringv1.Prometheus); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*monitoringv1.PrometheusRule); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*monitoringv1.ServiceMonitor); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*corev1.Namespace); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*apiextensionsv1beta1.CustomResourceDefinition); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*apiextensionsv1.CustomResourceDefinition); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*policyv1beta1.PodDisruptionBudget); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*netv1.NetworkPolicy); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*corev1.Endpoints); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*storkapi.ResourceTransformation); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*admissionregistrationv1.ValidatingWebhookConfiguration); ok {
+		for i := range specObj.Webhooks {
+			oldns := specObj.Webhooks[i].ClientConfig.Service.Namespace
+			specObj.Webhooks[i].ClientConfig.Service.Namespace = namespaceMapping[oldns]
+		}
+		return nil
 	}
 
 	return fmt.Errorf("unsupported object while setting namespace: %v", reflect.TypeOf(in))
 }
 
+func GetSpecNameKindNamepace(specObj interface{}) (string, string, string, error) {
+	if obj, ok := specObj.(*appsapi.Deployment); ok {
```

Comment:
Cam we use a loop instead?

Reply:
what do you mean?

Reply:
Don't know how to use a loop?

Reply:
Interesting now when I look at your code it looks like you don't even need a loop you can simply  it straight away a loop isn't needed as well here. 

```
if obj, ok := specObj.(*admissionregistrationv1.ValidatingWebhookConfiguration); ok {
		return obj.Name, obj.Kind, "", nil
} else {
      return obj.Name, obj.Kind, obj.Namespace, nil
} 

```

Reply:
If I'm understanding you correctly, you're confirming that a loop cannot be used; You're also asking me to use the snippet of code you provided;
But if `ok` is false, then won't `obj` be the default value? or `nil` (I'm unsure)?
I'm not sure if the `else` part of your snippet works as needed. The return statement I use also returns an error.

The code I used is actually copied from (as it's similar to) another previously written function in `k8s.go`, so we could refactor that code too, if we're refactoring this code.